orsopy/fileio/base.py

```diff
@@ -0,0 +1,174 @@
+"""
+Implementation of the base classes for the ORSO header.
+"""
+
+# author: Andrew R. McCluskey (arm61)
+
+import yaml
+
+
+def noop(self, *args, **kw):
+    pass
+
+
+yaml.emitter.Emitter.process_tag = noop
+
+
+def _repr(class_to_represent):
+    """
+    The representation object for all the Header sub-classes. This returns a
+    string in a yaml format which will be ORSO compatible.
+
+    Args:
+        class_to_represent (:py:class:`object`): The class to be represented.
+
+    Returns:
+        (:py:attr:`str`): A string representation.
+    """
+    cleaned_data = dict(
+        (k, v) for (k, v) in class_to_represent.__dict__.items(
+        ) if v is not None
+    )
+    return yaml.dump(cleaned_data, sort_keys=False)
+
+
+class Header:
+    """
+    The super class for all of the __repr__ items in the orso module
+    """
+
+    def __repr__(self):
+        """
+        The string representation for the Header class objects.
+        """
+        return _repr(self)
+
+
+class MagnetizationVector(Header):
+    """
+    A descriptor for the magnetization vector.
+    """
+    def __init__(self):
+        pass
+
+
+class ValueScalar(Header):
+    """
+    A single value with an unit.
+
+    Args:
+        magnitude (:py:attr:`float`): The value.
+        unit (:py:attr:`str`, optional): The unit. Optional,
+            defaults to :code:`'dimensionless'`.
+    """
+
+    def __init__(self, magnitude, unit="dimensionless"):
+        _check_unit(unit)
+        self.magnitude = magnitude
+        self.unit = unit
```

Comment:
I think units should be checked against the allowed ones. (In the class for generally being allowed and in later structures for the schema.)